src/main/scala/amba/axi4/Deinterleaver.scala

```diff
@@ -46,9 +47,12 @@ class AXI4Deinterleaver(maxReadBytes: Int)(implicit p: Parameters) extends LazyM
           }
         }
 
+        // This seems dubious. Are all these queues not actually used?
+        qs.foreach { _ := DontCare }
```

Comment:
@terpstra why is this OK (if I don't put this i get lots of firrtl errors about things being `VOID`)? At the very least it seems we should be defaulting the readys and valids to 0, not DontCare... down below they only get conditionally set which makes `chisel3` unhappy.

Reply:
Hmm. I don't understand this, either. The ready and valid seem to be driven unconditionally in the zip+foreach. Is it possible that the zip length is shorter than the 'qs' length?

Reply:
Hmm. I don't understand this, either. The ready and valid seem to be driven unconditionally in the zip+foreach. Is it possible that the zip length is shorter than the 'qs' length?

Reply:
i put a require that they be the same length, which did not fail, yet i still get the errors if I dont put in the `DontCare`.

Reply:
@terpstra did something change about `QueueIO`? in the case where you DON't put down a queue for `qs` and just put down a QueueIO, what makes the deq.bits := enq.bits ?

I think that's what its complaining about. I don't know why this would have worked before

Reply:
(talking about this line: https://github.com/chipsalliance/rocket-chip/blob/a344f719bd14db9a9e77b868eb645afba616f232/src/main/scala/amba/axi4/Deinterleaver.scala#L46)

Reply:
ah, the commit that added that line says "axi4: Deinterleaver need not make a Q for an unused AXI id". So presumably these wires are really `DontCare`-able, and we could put an assertion that their enq.fire == false?

Or I will plan to just wire them up so that deq := enq